### PR TITLE
Revert TableSelectionSet and IDbContextTransactionManager breaking changes.

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Part of #10545

### CODE REVIEW

File: src/EFCore/Extensions/TransactionsDatabaseFacadeExtensions.cs [hunk 1]
```diff
@@ -22,14 +22,28 @@ public static class TransactionsDatabaseFacadeExtensions
         /// <param name="databaseFacade"> The <see cref="DatabaseFacade" /> for the context.</param>
         /// <param name="transaction"> The transaction to be used. </param>
         public static void EnlistTransaction([NotNull] this DatabaseFacade databaseFacade, [CanBeNull] Transaction transaction)
-            => Check.NotNull(databaseFacade, nameof(databaseFacade)).GetService<IDbContextTransactionManager>().EnlistTransaction(transaction);
+        {
+            Check.NotNull(databaseFacade, nameof(databaseFacade));
+            if (databaseFacade.GetService<IDbContextTransactionManager>() is ITransactionManager transactionManager)
+            {
+                transactionManager.EnlistTransaction(transaction);
+            }
```
Comment (bricelam) [inline src/EFCore/Extensions/TransactionsDatabaseFacadeExtensions.cs]: Should we throw/warn if they call this and the provider doesn't support it?

Comment (AndriySvyryd) [reply src/EFCore/Extensions/TransactionsDatabaseFacadeExtensions.cs]: Yeah, we could throw by default, but noop for InMemory
@divega

Comment (divega) [reply src/EFCore/Extensions/TransactionsDatabaseFacadeExtensions.cs]: @AndriySvyryd AFAIR there is already an option flag for in-memory to ignore transaction calls. Should it only noop if that flag is enabled?

Comment (AndriySvyryd) [reply src/EFCore/Extensions/TransactionsDatabaseFacadeExtensions.cs]: They were always ignored previously. What would be the behavior if the flag is not set? Throw?

File: src/EFCore.Relational/Scaffolding/Internal/TableSelectionSet.cs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+using JetBrains.Annotations;
+using Microsoft.EntityFrameworkCore.Utilities;
+
+namespace Microsoft.EntityFrameworkCore.Scaffolding.Internal
+{
+    /// <summary>
+    ///     This API supports the Entity Framework Core infrastructure and is not intended to be used
+    ///     directly from your code. This API may change or be removed in future releases.
+    /// </summary>
+    [Obsolete("This is just to provide backward compatibility for 2.0 provider")]
```
Comment (divega) [inline src/EFCore.Relational/Scaffolding/Internal/TableSelectionSet.cs]: Nit: provider -> provider**s**

File: src/EFCore/Extensions/TransactionsDatabaseFacadeExtensions.cs (hunk 1 above)
Comment (divega) [reply src/EFCore/Extensions/TransactionsDatabaseFacadeExtensions.cs]: Actually, I think for the other transaction APIs we log `InMemoryEventId.TransactionIgnoredWarning`. Then you have to use ConfigureWarnings if you want to either throw or completely ignore it. Makes sense?

File: src/EFCore/Storage/ITransactionManager.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    /// <summary>
+    ///     <para>
+    ///         Manages the current <see cref="Transaction" />.
+    ///     </para>
+    ///     <para>
+    ///         This interface is typically used by database providers (and other extensions). It is generally
+    ///         not used in application code.
+    ///     </para>
+    /// </summary>
+    public interface ITransactionManager
```
Comment (ajcvickers) [inline src/EFCore/Storage/ITransactionManager.cs]: ITransactionEnlistmentManager?

Comment (AndriySvyryd) [reply src/EFCore/Storage/ITransactionManager.cs]: I have no objections. @divega?

**Pull request merged**